utils/swift_build_support/swift_build_support/cmake.py

```diff
@@ -177,3 +181,98 @@ def build_args(self):
                            '-jobs', str(jobs)]
 
         return build_args
+
+    # Determine the version of the installed CMake binary.
+    def installed_cmake_version(self, cmake_binary):
+        version = shell.capture([cmake_binary, '--version'], dry_run=False,
+                                    echo=True, optional=True)
+        if version is None:
+            return (0, 0, 0)
+
+        print(version)
+        x = re.findall(r'cmake version (\d+)\.(\d+)\.(\d+)', version.rstrip())
+        if len(x) == 1:
+            (c_major, c_minor, c_patch) = map(int, x[0])
+
+        return (c_major, c_minor, c_patch)
+
+    # Determine the version of the checked out CMake source.
+    def cmake_source_version(self, cmake_source_dir):
+        cmake_version_file = os.path.join(cmake_source_dir, 'Source',
+                                              'CMakeVersion.cmake')
+        print("cmake_version_file: %s" % cmake_version_file)
+        major = -1
+        minor = -1
+        patch = -1
+
+        file = open(cmake_version_file, "r")
+        for line in file.readlines():
+            m = re.findall(r'set\(CMake_VERSION_MAJOR (\d+)\)', line)
+            if len(m) == 1:
+                major = int(m[0])
+                next
+
+            m = re.findall(r'set\(CMake_VERSION_MINOR (\d+)\)', line)
+            if len(m) == 1:
+                minor = int(m[0])
+                next
+
+            m = re.findall(r'set\(CMake_VERSION_PATCH (\d+)\)', line)
+            if len(m) == 1:
+                patch = int(m[0])
+                next
+
+        if major == -1 or minor == -1 or patch == -1:
+            raise RuntimeError("Cant determine CMake version from %s"
+                                   % cmake_version_file)
+
+        return (major, minor, patch)
+
+    # Build CMake from source.
+    def build_cmake(self, source_root, build_root):
+        cmake_bootstrap = os.path.join(source_root, 'cmake', 'bootstrap')
+
+        if hasattr(self.args, 'build_script_impl_args'):
+            for opt in self.args.build_script_impl_args:
+                m = re.findall('--build-dir=(.*)', opt)
+                if len(m) == 1:
+                    build_root = m[0]
+
+        cmake_build_dir = os.path.join(build_root, 'cmake-%s' % self.args.host_target)
+        shell.call_without_sleeping(['mkdir', '-p', cmake_build_dir], echo=True)
```

Comment:
Please use `os.makedirs`.  This will fail on Windows.

Reply:
`os.makedirs()` wasn't available using the default python installed on Ubuntu18.04. (Python 2.7.15rc1). Is the python version on Windows guaranteed to have it? If so I could use something like:
```
if platform.System() == 'Linux':
  mkdir -p
else
  os.makedirs
```